proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java

```diff
@@ -511,7 +508,13 @@ public void run()
                             thisState = State.FINISHED;
                         }
                     }
-                } );
+                };
+
+                if ( ch.getHandle().eventLoop().inEventLoop() ) {
+                    nettyPostCall.run();
```

Comment:
If plugin doesn't change thread we can execute it immediately. Netty doesn't do it.